File: packages/executable-stories-junit5/src/main/kotlin/dev/executablestories/junit5/StoryStep.kt
package dev.executablestories.junit5

import com.fasterxml.jackson.annotation.JsonInclude

@JsonInclude(JsonInclude.Include.NON_NULL)
class StoryStep() {
    var id: String? = null
    var keyword: String? = null
    var text: String? = null
    var mode: String? = null
    var wrapped: Boolean? = null

    /**
     * Assertions attributable to this step.
     *
     * JUnit 5 has no assertion counter, so this is set only when the author
     * wraps a claim in `expect`/`fn("Then", ..)`. Null means unobserved, which
     * is not the same as zero.
     */
    var assertions: Int? = null
    var durationMs: Double? = null
    var docs: MutableList<DocEntry>? = null

    constructor(keyword: String, text: String) : this() {
        this.keyword = keyword
        this.text = text
    }

    fun addDoc(doc: DocEntry) {
        if (docs == null) {
            docs = mutableListOf()
        }
        docs!!.add(doc)
    }
}

File: packages/executable-stories-xunit/ExecutableStories.Xunit/StoryStep.cs
using System.Text.Json.Serialization;

namespace ExecutableStories.Xunit
{
    /// <summary>
    /// A single BDD step in a scenario.
    /// Matches the StoryStep definition in the raw-run schema.
    /// </summary>
    public class StoryStep
    {
        [JsonPropertyName("keyword")]
        public string Keyword { get; set; } = default!;

        [JsonPropertyName("text")]
        public string Text { get; set; } = default!;

        [JsonPropertyName("mode")]
        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        public string? Mode { get; set; }

        [JsonPropertyName("wrapped")]
        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        public bool? Wrapped { get; set; }

        /// <summary>
        /// Assertions attributable to this step.
        /// xUnit has no assertion counter, so this is set only when the author
        /// wraps a claim in <c>Expect</c>/<c>Fn("Then", ..)</c>. Null means
        /// unobserved, which is not the same as zero.
        /// </summary>
        public int? Assertions { get; set; }

        [JsonPropertyName("id")]
        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        public string? Id { get; set; }

        [JsonPropertyName("durationMs")]
        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        public double? DurationMs { get; set; }

        [JsonPropertyName("docs")]
        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingNull)]
        public List<DocEntry>? Docs { get; set; }
    }
}
